File: tree_primer/tree_primer__adjacency_matrix_input.py
# 実行時間 10ms
n = int(input())
g = [[0] * n for _ in [0] * n]

for _ in [0] * (n - 1):
    a, b = list(map(int, input().split()))

    a -= 1
    b -= 1
    g[a][b] = 1
    g[b][a] = 1
for i in range(n):
    print(*g[i])

File: tree_primer/tree_primer__adjacency_matrix_input.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int n = int.Parse( ReadLine() );
        byte[][] g = new byte[n][];

        foreach( int i in Range( 0, n ) ) {
            g[i] = new byte[n];
        }
        foreach( int _ in Range( 0, n - 1 ) ) {
            int[] a = ReadLine().Split().Select( x => int.Parse( x ) - 1 ).ToArray();

            g[a[0]][a[1]] = 1;
            g[a[1]][a[0]] = 1;
        }
        foreach( int i in Range( 0, n ) ) {
            WriteLine( string.Join( " ", g[i] ) );
        }
        return;
    }
}
